neutron/agent/linux/iptables_manager.py

```diff
@@ -199,6 +231,35 @@ def remove_chain(self, name, wrap=True):
         self.rules = [r for r in self.rules
                       if jump_snippet not in r.rule]
 
+    def remove_chain_by_tag(self, name, tag, wrap=True):
```

Comment:
Initially I did change the original method by just assuming that the chain name is used for tagging the rules. I decided that it would be better to separate the tag from the chain name so that it is clear what is being done.